### PR TITLE
Karmada Highly Available Topology Overview

### PR DESCRIPTION
**Pull request by Affan-7**
**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:
This PR introduces an overview to HA architecture of karmada, so users get aware about Karmada HA.

**Which issue(s) this PR fixes**:
Related to: #418
Task 1 out of 5

**Special notes for your reviewer**:
@RainbowMango can you please review the architecture in this draft.
Feel free to add some suggestions. I will move forward when everything looks good to you.

Newly added page: https://deploy-preview-443--karmada.netlify.app/docs/next/installation/ha-installation/

### CODE REVIEW

**Review by RainbowMango — COMMENTED** [review]
/assign

**Comment by RainbowMango** [comment]
@Affan-7 , Thank you very much for doing this. That's awesome. 

I think Karmada's HA Topology is quite similar to Kubernetes. So can we describe it as [this document](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/ha-topology/)?
- Stacked etcd topology
- External etcd topology

File: docs/installation/ha-installation.md [hunk 1]
```diff
@@ -2,10 +2,58 @@
...
+In Karmada HA a member cluster can get registered to the multiple Karmada control planes.
+So even if a Karmada control plan goes down, you can still manage your clusters
+using the other Karmada control planes.
+
+Here is a diagram of Karmada HA architecture.
+
 ![Karmada HA 1](../resources/general/ha-karmada-1.png)
-![Karmada HA 2](../resources/general/ha-karmada-2.png)
\ No newline at end of file
+![Karmada HA 2](../resources/general/ha-karmada-2.png)
```
Comment (RainbowMango) [inline docs/installation/ha-installation.md:17]: These words and pictures seem a little bit confusing. I guess we can remove them as they are fully explained in the below content.

> In Karmada HA a member cluster can get registered to the multiple Karmada control planes.

The `register` in Karmada usually means the action a cluster joins to control plane, usually a cluster just needs to register to one control plane.

> Karmada HA 1
Karmada HA 2

The two pictures show the 3 ETCDs are completely independent, which conflicts with the below `Stacked etcd topology`.

File: docs/installation/ha-installation.md [hunk 2]
```diff
@@ -2,10 +2,58 @@
...
+
+## Stacked etcd topology
+
+A stacked HA cluster is a topology where the distributed data storage cluster provided by etcd is
+stacked on top of the Karamada control plane nodes.
+
+Each control plane node runs an instance of the Karmada API server, Karmada scheduler, and Karmada controller manager. The Karmada API server can communicate with the multiple member clusters, and
+these member clusters can be registered to the multiple Karmada API servers.
+
+Each Karmada control plane node creates a local etcd database and this etcd database communicates only with the Karmada API server of that node. The same applies to the local Karmada controller manager and Karmada scheduler instances.
```
Comment (RainbowMango) [inline docs/installation/ha-installation.md:39]:
```suggestion
Each Karmada control plane node creates a local etcd member and this etcd member communicates only with the Karmada API server of that node. The same applies to the local Karmada controller manager and Karmada scheduler instances.
```

File: docs/installation/ha-installation.md [hunk 3]
```diff
@@ -2,10 +2,58 @@
...
+
+This topology couples the control planes and etcd databases on the same nodes. It is simpler to set up than a cluster with external etcd nodes, and simpler to manage for replication.
+
+However, a stacked cluster runs the risk of failed coupling. If one node goes down, both an etcd database and a Karmada control plane instance are lost, and redundancy is compromised. You can mitigate this risk by adding more control plane nodes.
```
Comment (RainbowMango) [inline docs/installation/ha-installation.md:43]:
```suggestion
However, a stacked cluster runs the risk of failed coupling. If one node goes down, both an etcd member and a Karmada control plane instance are lost, and redundancy is compromised. You can mitigate this risk by adding more control plane nodes.
```

**Review by RainbowMango — COMMENTED** [review]
Looks awesome, just some nits, otherwise LGTM.

**Review by RainbowMango — COMMENTED** [review]
Could you please [squash your commits](https://karmada.io/docs/contributor/github-workflow#squash-commits)? Then things will be all set, thanks, well done.

By the way, which tools did you use to draw the excellent picture? Please add the source file if possible,.

**Comment by Affan-7** [comment]
Thank you for collaboration @RainbowMango, I appreciate your contributions to the open-source.

I use www.canva.com for creating diagrams and graphics. I discovered this tool while I was creating some graphics for my tech blog. It's very handy and easy to use. Here are the source files.

https://www.canva.com/design/DAFrH6RfJp4/ISaw242zbJxnH5HOUElF2g/edit?utm_content=DAFrH6RfJp4&utm_campaign=designshare&utm_medium=link2&utm_source=sharebutton

https://www.canva.com/design/DAFrH9x9WiI/JURdm-FBF0P8hlV-7vSzCw/edit?utm_content=DAFrH9x9WiI&utm_campaign=designshare&utm_medium=link2&utm_source=sharebutton

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**